src/blocktridiag.jl

```diff
@@ -0,0 +1,243 @@
+#=======================
+ BlockTridiagMatrix
+=======================#
+
+"""
+    BlockTridiagMatrix{T, R <: AbstractMatrix{T}} <: AbstractBlockArray{T, N}
+
+A `BlockTridiagMatrix` is a block tridiagonal matrix where each block is stored contiguously.
+This means that insertions and retrieval of blocks
+can be very fast and non allocating since no copying of data is needed.
+
+In the type definition, `R` defines the array type that each block has,
+for example `Matrix{Float64}.
+"""
+struct BlockTridiagMatrix{T, R <: AbstractMatrix{T}} <: AbstractBlockMatrix{T}
+    diagl::Vector{R}
+    lower::Vector{R}
+    upper::Vector{R}
+    block_sizes::BlockSizes{2}
+end
+
+# Auxilary outer constructors
+function BlockTridiagMatrix{T, R <: AbstractArray{T}
+                           }(diagl::Vector{R},
```

Comment:
In my opinion, these linebreaks before the `}` looks a bit weird and I haven't seen Julia code using it before.